dashdotdb/services/serviceslometrics.py

```diff
@@ -8,19 +8,32 @@
 from dashdotdb.models.dashdotdb import Namespace
 from dashdotdb.models.dashdotdb import ServiceSLO
 from dashdotdb.models.dashdotdb import SLIType
+from dashdotdb.models.dashdotdb import SLODoc
 from dashdotdb.services import DataTypes
 from dashdotdb.controllers.token import (TOKEN_NOT_FOUND_CODE,
                                          TOKEN_NOT_FOUND_MSG)
 
 
+class ServiceSLOMetricsInput:
+    def __init__(self):
+        self.cluster: str = None
+        self.namespace: str = None
+        self.sli_type: str = None
+        self.slo_doc: str = None
+        self.name: str = None
+
+
 class ServiceSLOMetrics:
-    def __init__(self, cluster=None, namespace=None, sli_type=None, name=None):
+    def __init__(self,
+                 input_props: ServiceSLOMetricsInput =
+                 ServiceSLOMetricsInput()):
```

Comment:
The behavior for default args like this might not be [what you expect](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments). You can generally avoid unexpected behavior by either making this arg mandatory or default it to `None` and then check for `None` and create it in `__init__`.